R/plot.R

```diff
@@ -2,62 +2,145 @@
 ### This file contains graphing functions for plotting the output from the ML_functions
 #############
 
-plot_scatter <- function(kfold_results) {
-  ggplot2::ggplot(kfold_results, ggplot2::aes(x=true, y=predicted)) +
+check_ids_match <- function(dfs) {
+  ids <- lapply(dfs, function(df) df$id)
+  all_equal <- lapply(ids, function(id_list) id_list == ids[[1]])
+  all(unlist(all_equal))
+}
+
+check_trues_match <- function(dfs) {
+  trues <- lapply(dfs, function(df) df$true)
+  all_equal <- lapply(trues, function(trues_list) all.equal(trues_list, trues[[1]]))
+  all(unlist(all_equal))
+}
+
+join_dfs <- function(dfs, keep_fold=FALSE) {
+  if(length(dfs) != 1) {
+    stopifnot(check_ids_match(dfs))
+    stopifnot(check_trues_match(dfs))
```

Comment:
Aren't id's getting created after the prediction function is happening in the cv function? If so are we sure that matching then means what we want it to mean? I guess the combination of both ids and true matching means we should be safe